### PR TITLE
test(util): add options & choices tests

### PR DESCRIPTION
**Pull request by rigormorrtiss**
image

### CODE REVIEW

**Comment by itohatweb** [comment]
pls. remove the `validates` in the names

**Review by ayntee — APPROVED** [review]

File: tests/util/utils.ts [hunk 1]
```diff
@@ -79,7 +80,7 @@ Deno.test({
 });
 
 Deno.test({
-  name: "[utils] validateSlashCommands(): validates name",
+  name: "[utils] validateSlashCommands(): validates application command name",
```
Comment (itohatweb) [inline tests/util/utils.ts:83]:
```suggestion
  name: "[utils] validateSlashCommands(): application command name",
```

File: tests/util/utils.ts [hunk 2]
```diff
@@ -95,7 +96,8 @@ Deno.test({
 });
 
 Deno.test({
-  name: "[utils] validateSlashCommands(): validates description",
+  name:
+    "[utils] validateSlashCommands(): validates application command description",
```
Comment (itohatweb) [inline tests/util/utils.ts:100]:
```suggestion
    "[utils] validateSlashCommands(): application command description",
```

File: tests/util/utils.ts [hunk 3]
```diff
@@ -126,3 +128,161 @@ Deno.test({
     }]);
   },
 });
+
+Deno.test({
+  name: "[utils] validateSlashCommands(): validates number of option choices",
```
Comment (itohatweb) [inline tests/util/utils.ts:133]:
```suggestion
  name: "[utils] validateSlashCommands(): number of option choices",
```

File: tests/util/utils.ts [hunk 4]
```diff
@@ -126,3 +128,161 @@ Deno.test({
...
+        type: DiscordApplicationCommandOptionTypes.String,
+        description: "The description of the option.",
+        choices: [choice],
+      }],
+    }]);
+  },
+});
+
+Deno.test({
+  name: "[utils] validateSlashCommands(): validates option name",
```
Comment (itohatweb) [inline tests/util/utils.ts:165]:
```suggestion
  name: "[utils] validateSlashCommands(): option name",
```

File: tests/util/utils.ts [hunk 5]
```diff
@@ -126,3 +128,161 @@ Deno.test({
...
+        name: "optionname",
+        description: "The description of the option.",
+        type: DiscordApplicationCommandOptionTypes.String,
+      }],
+    }]);
+  },
+});
+
+Deno.test({
+  name: "[utils] validateSlashCommands(): validates option description",
```
Comment (itohatweb) [inline tests/util/utils.ts:189]:
```suggestion
  name: "[utils] validateSlashCommands(): option description",
```

File: tests/util/utils.ts [hunk 6]
```diff
@@ -126,3 +128,161 @@ Deno.test({
...
+        name: "optionname",
+        description: "The description of the option.",
+        type: DiscordApplicationCommandOptionTypes.String,
+      }],
+    }]);
+  },
+});
+
+Deno.test({
+  name: "[utils] validateSlashCommands(): validates the option choice name",
```
Comment (itohatweb) [inline tests/util/utils.ts:213]:
```suggestion
  name: "[utils] validateSlashCommands(): the option choice name",
```

File: tests/util/utils.ts [hunk 7]
```diff
@@ -126,3 +128,161 @@ Deno.test({
...
+          name: "choicename",
+          value: "choicevalue",
+        }],
+      }],
+    }]);
+  },
+});
+
+Deno.test({
+  name: "[utils] validateSlashCommands(): validates option choice value",
```
Comment (itohatweb) [inline tests/util/utils.ts:245]:
```suggestion
  name: "[utils] validateSlashCommands(): option choice value",
```

**Pull request merged**